pyspecdata/core.py

```diff
@@ -4565,99 +4635,100 @@ def nnls(self, dimname, newaxis_dict, kernel_func, l=0):
-            fit_axis = newaxis_dict[fitdim_name]
-        if tuple_syntax:
-            fit_axis1 = nddata(fit_axis1,fitdim_name1)
-            fit_axis2 = nddata(fit_axis2,fitdim_name2)
-            data_axis1 = self.fromaxis(dimname[0])
-            data_axis2 = self.fromaxis(dimname[1])
-            data_axis1.squeeze()
-            data_axis2.squeeze()
-            data_axis1,fit_axis1 = data_axis1.aligndata(fit_axis1)
-            data_axis2,fit_axis2 = data_axis2.aligndata(fit_axis2)
-            K1 = kernel_func[0](data_axis1,fit_axis1).squeeze()
-            K1_ret = K1
-            logger.debug(strm('K1 dimlabels',K1.dimlabels,'and raw shape',K1.data.shape))
-            K2 = kernel_func[1](data_axis2,fit_axis2).squeeze()
-            K2_ret = K2
-            logger.debug(strm('K2 dimlabels',K2.dimlabels,'and raw shape',K2.data.shape))
-            # SVD and truncation of kernels
-            U1,S1,V1 = np.linalg.svd(K1.data,full_matrices=False)
-            U2,S2,V2 = np.linalg.svd(K2.data,full_matrices=False)
-            logger.debug(strm('Uncompressed SVD K1:',[x.shape for x in (U1,S1,V1)]))
-            logger.debug(strm('Uncompressed SVD K2:',[x.shape for x in (U2,S2,V2)]))
-            default_cut = 1e-2
-            s1 = where(S1 > default_cut)[0][-1]
+        fit_dimnames = [j.dimlabels[0] for j in newaxis_dict]
+        fit_dimaxes = [j.getaxis(fit_dimnames[j_idx]) for j_idx,j in enumerate(newaxis_dict) if j.getaxis(fit_dimnames[j_idx]) is not None]
+        #    else:
+        #        fit_dimaxes = [j.data]
+        fit_axes = [nddata(fit_dimaxes[j],fit_dimnames[j]) for j in range(len(dimname))]
+        data_axes = [self.fromaxis(dimname[j]) for j in range(len(dimname))]
+        for j in range(len(dimname)):
+            data_axes[j].squeeze()
+        for j in range(len(dimname)):
+            data_axes[j],fit_axes[j] = data_axes[j].aligndata(fit_axes[j])
+        # note I specified K1_ret and K2_ret for returning kernels as properties of the nddata
+        kernels = [kernel_func[j](data_axes[j],fit_axes[j]).squeeze() for j in range(len(dimname))]
+        logger.debug(strm('K%d dimlabels'%j,kernels[j].dimlabels,'and raw shape',kernels[j].data.shape) for j in range(len(dimname)))
+        svd_return = []
+        for j in range(len(dimname)):
+            svd_return.append(np.linalg.svd(kernels[j].data,full_matrices=False))
+        U1 = (svd_return[0])[0]
+        S1 = (svd_return[0])[1]
+        V1 = (svd_return[0])[2]
+        default_cut = 1e-2
+        s1 = where(S1 > default_cut)[0][-1]
+        U1 = U1[:,0:s1]
+        S1 = S1[0:s1]
+        V1 = V1[0:s1,:]
+        S1 = S1*eye(s1)
+        logger.debug(strm('Compressed SVD of K1:',[x.shape for x in (U1,S1,V1)]))
+        #{{{ prepping 2D
+        if twoD:
+            U2 = (svd_return[1])[0]
+            S2 = (svd_return[1])[1]
+            V2 = (svd_return[1])[2]
             s2 = where(S2 > default_cut)[0][-1]
```

Comment:
@aabeaton : just checking that this branch is the most recent version of the code.  I thought maybe we landed on a list of U matrices, list of V matrices, etc. (list can have 1 or 2 elements) -- or is this just done for the kernel, or not at all, or is this not eh most recent version of the code?